packages/core/content-manager/server/services/utils/configuration/layouts.js

```diff
@@ -7,9 +7,16 @@ const { isListable, hasEditableAttribute, hasRelationAttribute } = require('./at
 const DEFAULT_LIST_LENGTH = 4;
 const MAX_ROW_SIZE = 12;
 
-const isAllowedFieldSize = (type, size) => {
+const isAllowedFieldSize = (type, size, customFieldUid = null) => {
   const { getFieldSize } = getService('field-sizes');
-  const fieldSize = getFieldSize(type);
+  let fieldSize = getFieldSize(type);
+
+  if (customFieldUid) {
+    const customField = strapi.container.get('custom-fields').get(customFieldUid);
+    if (customField.inputSize) {
+      fieldSize = customField.inputSize;
+    }
+  }
```

Comment:
- I think sole responsibility of this function should be to return a boolean, not set a value. It seems like the custom field value should already be set when we call getFieldSize.
- It looks like we have a function here that sets the user provided field size for all custom fields. Why do we need this addition here? https://github.com/strapi/strapi/blob/ee14338b977d4d3b4271a2d61fe3807ef4bfff8c/packages/core/content-manager/server/services/field-sizes.js#L80-L91
Which is called in bootstrap:
https://github.com/strapi/strapi/blob/207938b001795705e278ef1fcfd6d32534590d20/packages/core/content-manager/server/bootstrap.js#L14